src/main/java/core/basesyntax/service/strategy/TypeStrategyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.FruitRecord;
+import core.basesyntax.service.strategy.strategyimpl.OperationHandler;
+import java.util.Map;
+
+public class TypeStrategyImpl implements TypeStrategy {
+    private final Map<FruitRecord.Operation, OperationHandler> operationHandlers;
+
+    public TypeStrategyImpl(Map<FruitRecord.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getType(FruitRecord.Operation operation) {
```

Comment:
```suggestion
    public OperationHandler getHandler(FruitRecord.Operation operation) {
```

Reply:
not fixed
getType isn't reflecting what the method is doing IMO
you return from OperationHandler from the method, but it's called getType